tests/Builder/DatagridBuilderTest.php

```diff
@@ -11,69 +11,159 @@
-     */
-    protected $typeGuesser;
-
     /**
      * @var DatagridBuilder
      */
-    protected $datagridBuilder;
-    /**
-     * @var FormFactoryInterface|\PHPUnit_Framework_MockObject_MockObject
-     */
+    private $datagridBuilder;
+    private $typeGuesser;
     private $formFactory;
+    private $filterFactory;
+    private $admin;
+    private $modelManager;
+
+    protected function setUp()
+    {
+        $this->formFactory = $this->prophesize(FormFactoryInterface::class);
+        $this->filterFactory = $this->prophesize(FilterFactoryInterface::class);
+        $this->typeGuesser = $this->prophesize(TypeGuesserInterface::class);
+
+        $this->datagridBuilder = new DatagridBuilder(
+            $this->formFactory->reveal(),
+            $this->filterFactory->reveal(),
+            $this->typeGuesser->reveal()
+        );
+
+        $this->admin = $this->prophesize(AbstractAdmin::class);
+        $this->modelManager = $this->prophesize(ModelManager::class);
+
+        $this->admin->getClass()->willReturn('FakeClass');
+        $this->admin->getModelManager()->willReturn($this->modelManager->reveal());
+        $this->admin->attachAdminClass(Argument::cetera())->willReturn();
+        $this->admin->addFilterFieldDescription(Argument::cetera())->willReturn();
+    }
 
     /**
-     * @var FilterFactoryInterface|\PHPUnit_Framework_MockObject_MockObject
+     * @dataProvider getBaseDatagridData
      */
-    private $filterFactory;
+    public function testGetBaseDatagrid($pagerType, $pager)
+    {
+        $proxyQuery = $this->prophesize(ProxyQueryInterface::class);
+        $fieldDescription = $this->prophesize(FieldDescriptionCollection::class);
+        $formBuilder = $this->prophesize(FormBuilderInterface::class);
 
-    protected function setUp()
+        $this->admin->getPagerType()->willReturn($pagerType);
+        $this->admin->createQuery()->willReturn($proxyQuery->reveal());
+        $this->admin->getList()->willReturn($fieldDescription->reveal());
+
+        $this->modelManager->getIdentifierFieldNames(Argument::any())->willReturn(['id']);
+
+        $this->formFactory->createNamedBuilder(Argument::cetera())->willReturn($formBuilder->reveal());
+
+        $this->assertInstanceOf(
+            Datagrid::class,
+            $datagrid = $this->datagridBuilder->getBaseDatagrid($this->admin->reveal())
+        );
+        $this->assertInstanceOf($pager, $datagrid->getPager());
+    }
+
+    public function getBaseDatagridData()
     {
-        $this->formFactory = $this->createMock(FormFactoryInterface::class);
-        $this->filterFactory = $this->createMock(FilterFactoryInterface::class);
-        $this->typeGuesser = $this->createMock(TypeGuesserInterface::class);
+        return [
+            [
+                Pager::TYPE_SIMPLE,
+                SimplePager::class,
+            ],
+            [
+                Pager::TYPE_DEFAULT,
+                Pager::class,
+            ],
+        ];
```

Comment:
It's good practice to specify a description for each item of a data provider as its key
```
return [
   'simple' => [
…
]
]
```

Reply:
will add keys, any idea about the issue with the test, what @dmarkowicz suggested make sense but would that be a BC break if we move to 0.9?

Reply:
Increase dependencies is not a BC break, IMO let's do it